ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/dialogs/DialogFactory.java

```diff
@@ -207,6 +207,40 @@ InputDialog createInputDialog(@NotNull @Assisted("title") String title,
                                   @Nullable CancelCallback cancelCallback);
 
     /**
+     * Create an input dialog with the specified initial value.
+     * <p/>
+     * The {@code initialValue} may be pre-selected. Selection begins
+     * at the specified {@code selectionStartIndex} and extends to the
+     * character at index {@code selectionLength}.
+     *
+     * @param title
+     *         the window title
+     * @param label
+     *         the label of the input field
+     * @param initialValue
+     *         the value used to initialize the input
+     * @param selectionStartIndex
+     *         the beginning index of the {@code initialValue} to select, inclusive
+     * @param selectionLength
+     *         the number of characters of the {@code initialValue} to be selected
+     * @param okButtonLabel
+     *         label for OK button
+     * @param inputCallback
+     *         the callback used on OK
+     * @param cancelCallback
+     *         the callback used on cancel
+     * @return an {@link InputDialog} instance
+     */
+    InputDialog createInputDialog(@NotNull @Assisted("title") String title,
```

Comment:
We can leave it for the future. Please give a look at the create folder popup. There are two buttons Cancel and OK.
It would be better to have Cancel and Create instead.